felt/src/lib.rs

```diff
@@ -875,20 +878,20 @@ assert_felt_impl!(Felt252);
 #[cfg(test)]
 mod test {
     use super::*;
+    use crate::arbitrary::{any_felt252, nonzero_felt252};
     use core::cmp;
     use proptest::prelude::*;
 
-    const FELT_PATTERN: &str = "(0|[1-9][0-9]*)";
-    const FELT_NON_ZERO_PATTERN: &str = "[1-9][0-9]*";
-
     proptest! {
+        #![proptest_config(ProptestConfig::with_cases(100000))]
         #[test]
-        #[allow(deprecated)]
         #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
-        // Property-based test that ensures, for 100 felt values that are randomly generated each time tests are run, that a new felt doesn't fall outside the range [0, p].
-        // In this and some of the following tests, The value of {x} can be either [0] or a very large number, in order to try to overflow the value of {p} and thus ensure the modular arithmetic is working correctly.
-        fn new_in_range(ref x in any::<[u8; 40]>()) {
-            let x = &Felt252::from_bytes_be(x);
+        // Property-based test that ensures, for 100 felt values that are randomly generated
+        // each time tests are run, that a new felt doesn't fall outside the range [0, p].
+        // In this and some of the following tests, The value of {x} can be either [0] or a
+        // very large number, in order to try to overflow the value of {p} and thus ensure the
+        // modular arithmetic is working correctly.
+        fn new_in_range(ref x in any_felt252()) {
```

Comment:
Careful here. We want specifically to test inputs that are out of range in this case, to confirm the constructor adjusts them. So, here we should use an arbitrary `BigUint` or `BigInt` (so we can also work with signed numbers) rather than a "correct" `Felt252`.

Reply:
Thanks for the input. I completely overlooked these types of tests. I'll search for these cases and fix them.